### PR TITLE
Support MPI executables (on a single process)

### PR DESCRIPTION
**Pull request by ltalirz**
This builds on #45 and makes it possible to run executables through `mpirun`, but on a single-process only(!).

Previously, when the mock executable would detect that an input is not present in the cache, it would
 * replace its name in the submit script with the fallback executable
 * run the submit script (again) using bash

This approach obviously does not work in conjunction with MPI, since it ends up calling `mpirun` twice: once for running the mock executable, and once again _inside_ the mock executable when it calls the submit script again.

This PR changes the mock executable to simply call the fallback executable directly, using the command line arguments that were provided.
Environment variables are forwarded to the subprocess automatically using `subprocess.call`.

This works for running with `-np 1`, but it still does not solve running with more than one process (one should probably add a check).

### CODE REVIEW

**Comment by ltalirz** [comment]
Reviving this pull request, as it is used in `aiida-lsmo` required by @mpougin

@greschd do you see any obvious issues with this approach?

File: aiida_testing/mock_code/_cli.py [hunk 1]
```diff
@@ -46,8 +46,7 @@ def run() -> None:
             sys.exit("No existing output, and no executable specified.")
 
         # replace executable path in submit file and run calculation
```
Comment (greschd) [inline aiida_testing/mock_code/_cli.py]:
```suggestion
```

**Review by greschd — DISMISSED** [review]

**Comment by greschd** [comment]
LGTM, apart from the suggestion above (remove outdated comment).

**Comment by ltalirz** [comment]
Thanks for the quick review @greschd, much appreciated!

**Pull request merged**